packages/@eventual/cli/src/commands/local.ts

```diff
@@ -294,6 +324,20 @@ export const local = (yargs: Argv) =>
             : ""
         }`
       );
+
+      let saved = false;
+      const save = () => {
+        if (!saved) {
+          persistanceStore.save();
+          saved = true;
+        }
+      };
+      process.on("exit", () => {
+        save();
+      });
+      process.on("SIGINT", () => {
+        save();
+      });
```

Comment:
I wonder if there are any race conditions? Eg if the workflow keeps running?

Also, are you sure you can reliably run async code on exit?